graphql/schemabuilder/pagination.go

```diff
@@ -452,6 +557,96 @@ func castSlice(slice interface{}) []interface{} {
 	return ret
 }
 
+// buildEmbeddedArgParser when the user embeds in the pagination args.
+func (sb *schemaBuilder) buildEmbeddedPaginatedArgParser(typ reflect.Type) (*argParser, graphql.Type, error) {
+	fields := make(map[string]argField)
+
+	argType := &graphql.InputObject{
+		Name:        typ.Name(),
+		InputFields: make(map[string]graphql.Type),
+	}
+	pagArgIndex := 0
+	argType.Name += "_InputObject"
+	for i := 0; i < typ.NumField(); i++ {
+		field := typ.Field(i)
+
+		// The field which is of type interface should only be one and will be used to parse the
+		// original function args.
+		if field.Type.Kind() == reflect.Interface {
+			continue
+		}
+		if field.Type == reflect.TypeOf(PaginationArgs{}) {
+			pagArgIndex = i
+			continue
+		}
+
+		name := makeGraphql(field.Name)
+
+		var parser *argParser
+		var fieldArgTyp graphql.Type
+
+		parser, fieldArgTyp, err := sb.makeArgParser(field.Type)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		argType.InputFields[name] = fieldArgTyp
+		fields[name] = argField{
+			field:  field,
+			parser: parser,
+		}
+	}
+
+	pagArgParser, pagArgType, err := sb.makeStructParser(reflect.TypeOf(PaginationArgs{}))
+	if err != nil {
+		return nil, nil, err
+	}
+	pagObj, ok := pagArgType.(*graphql.InputObject)
+	if !ok {
+		panic("failed to cast paginated args to an input object")
+	}
+	for name, objField := range pagObj.InputFields {
+		if _, ok := argType.InputFields[name]; ok {
+			return nil, nil, fmt.Errorf("These arg names are restricted: First, After, Last and Before")
```

Comment:
nit: start error with lowercase

Reply:
ping on this

Reply:
oh, i see this is done in the third commit. it'd be nice to squash that down to here.